core/shared/src/main/scala/zio/Hub.scala

```diff
@@ -171,13 +171,15 @@ object Hub {
           else ZIO.succeed(hub.size())
         }
       def subscribe(implicit trace: Trace): ZIO[Scope, Nothing, Dequeue[A]] =
-        ZIO.acquireRelease {
-          makeSubscription(hub, subscribers, strategy).tap { dequeue =>
-            scope.addFinalizer(dequeue.shutdown)
-          }
-        } { dequeue =>
-          dequeue.shutdown
-        }
+        ZIO.acquireReleaseExit {
+          for {
+            child   <- scope.fork
```

Comment:
I'm not sure if this will help with the growing release map - `scope.fork` under the hood seems to also add a finalizer to release map

Reply:
It removes the finalizer when the scope is closed.

Reply:
ok, now i noticed how it works, it should be ok - sorry for bothering you

Reply:
No worries! Thanks for raising!